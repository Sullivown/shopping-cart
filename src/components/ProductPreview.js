import React from 'react';
import { useNavigate } from 'react-router-dom';

function ProductPreview(props) {
	let navigate = useNavigate();
	return (
		<section className='product-preview'>
			<img src='#' alt={props.product.title} />
			<h3>{props.title}</h3>
			<button
				type='button'
				onClick={() => navigate(`/shop/${props.product.id}`)}
			>
				View Item
			</button>
			<button>Add to Cart</button>
		</section>
	);
}

export default ProductPreview;
