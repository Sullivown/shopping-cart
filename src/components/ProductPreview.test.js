import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import ProductPreview from './ProductPreview';

const testProduct = {
	id: 1,
	title: '3D Glasses',
	price: 2.99,
	image: jest.mock('../assets/images/product-images/1.png'),
};

describe('Test ProductPreview Component', () => {
	const addToCartMock = jest.fn();
	it('renders correctly', () => {
		const productPreviewTestRenderer = TestRenderer.create(
			<Router>
				<ProductPreview
					key={testProduct.id}
					product={testProduct}
					addToCart={addToCartMock}
				/>
			</Router>
		).toJSON();

		expect(productPreviewTestRenderer).toMatchSnapshot();
	});

	it('Add to cart clicked once', () => {
		render(
			<Router>
				<ProductPreview
					key={testProduct.id}
					product={testProduct}
					addToCart={addToCartMock}
				/>
			</Router>
		);

		const addToCartButton = screen.getByRole('button', {
			name: 'Add to Cart',
		});

		userEvent.click(addToCartButton);

		expect(addToCartMock).toHaveBeenCalled();
	});

	it('Add to cart clicked multiple times', () => {
		render(
			<Router>
				<ProductPreview
					key={testProduct.id}
					product={testProduct}
					addToCart={addToCartMock}
				/>
			</Router>
		);

		const addToCartButton = screen.getByRole('button', {
			name: 'Add to Cart',
		});

		userEvent.click(addToCartButton);
		userEvent.click(addToCartButton);
		userEvent.click(addToCartButton);

		expect(addToCartMock).toHaveBeenCalledTimes(3);
	});
});
