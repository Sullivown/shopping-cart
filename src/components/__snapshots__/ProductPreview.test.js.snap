// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductPreview Component renders correctly 1`] = `
<section
  className="product-preview"
>
  <img
    alt="3D Glasses"
    onClick={[Function]}
    src={
      Object {
        "advanceTimersByTime": [Function],
        "advanceTimersToNextTimer": [Function],
        "autoMockOff": [Function],
        "autoMockOn": [Function],
        "clearAllMocks": [Function],
        "clearAllTimers": [Function],
        "createMockFromModule": [Function],
        "deepUnmock": [Function],
        "disableAutomock": [Function],
        "doMock": [Function],
        "dontMock": [Function],
        "enableAutomock": [Function],
        "fn": [Function],
        "genMockFromModule": [Function],
        "getRealSystemTime": [Function],
        "getTimerCount": [Function],
        "isMockFunction": [Function],
        "isolateModules": [Function],
        "mock": [Function],
        "mocked": [Function],
        "requireActual": [Function],
        "requireMock": [Function],
        "resetAllMocks": [Function],
        "resetModules": [Function],
        "restoreAllMocks": [Function],
        "retryTimes": [Function],
        "runAllImmediates": [Function],
        "runAllTicks": [Function],
        "runAllTimers": [Function],
        "runOnlyPendingTimers": [Function],
        "setMock": [Function],
        "setSystemTime": [Function],
        "setTimeout": [Function],
        "spyOn": [Function],
        "unmock": [Function],
        "unstable_mockModule": [Function],
        "useFakeTimers": [Function],
        "useRealTimers": [Function],
      }
    }
  />
  <h3
    onClick={[Function]}
  >
    3D Glasses
  </h3>
  <p>
    Â£2.99
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    View Item
  </button>
  <button
    onClick={[Function]}
  >
    Add to Cart
  </button>
</section>
`;
